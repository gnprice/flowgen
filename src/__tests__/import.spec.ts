import { compiler, beautify } from "..";
import "../test-matchers";

it("should handle dynamic imports", () => {
  const ts = `
type A = import('react');
type B = import('react').ComponentType<{}>;
type C = import('react').ReactNode;
type D = import('zlib').Zlib;
type Z = number;
`;
  const result = compiler.compileDefinitionString(ts, { quiet: true });
  expect(beautify(result)).toMatchSnapshot();
  expect(result).toBeValidFlowTypeDeclarations();
});

it("should handle imports from odd names", () => {
  const ts = `
type A = import('..');
type B = import('@!-/./');
`;
  const result = compiler.compileDefinitionString(ts, { quiet: true });
  expect(beautify(result)).toMatchSnapshot();
  // expect(result).toBeValidFlowTypeDeclarations(); // would need actual modules at those names
});
