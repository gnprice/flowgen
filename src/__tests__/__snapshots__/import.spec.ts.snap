// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle dynamic imports 1`] = `
"import * as $Flowgen$Import$react from \\"react\\";
import * as $Flowgen$Import$zlib from \\"zlib\\";
declare type A = typeof $Flowgen$Import$react;
declare type B = $Flowgen$Import$react.ComponentType<{ ... }>;
declare type C = $Flowgen$Import$react.Node;
declare type D = $Flowgen$Import$zlib.Zlib;
declare type Z = number;
"
`;

exports[`should handle imports from odd names 1`] = `
"import * as $Flowgen$Import$_46_46 from \\"..\\";
import * as $Flowgen$Import$_64_33_45_47_46_47 from \\"@!-/./\\";
declare type A = typeof $Flowgen$Import$_46_46;
declare type B = typeof $Flowgen$Import$_64_33_45_47_46_47;
"
`;
